tests/post/steps/test_example_solution.py

```diff
@@ -0,0 +1,398 @@
+def is_absent_solution(host, name, k8s_client):
+    with host.sudo():
+        if name in host.mount_point.get_mountpoints():
+            assert False, 'Found Solution mountpoint {}'.format(name)
+    return name
+
+
+@given(parsers.parse("no Solution environment '{environment}' is available"))
+def is_absent_solution_environment(host, environment, k8s_client):
+    solution_environment = get_environment(k8s_client, environment)
+    if solution_environment is not None:
+        pytest.fail("Found Solution environment {}".format(environment))
+    return environment
+
+
+@given(parsers.parse("the Solution Configuration file is absent"))
+def is_absent_solution_config(host):
+    with host.sudo():
+        if host.file(SOLUTION_CONFIGURATION_FILE).exists:
+            assert False, 'Solution Config file exist at path {}'.format(
+                        SOLUTION_CONFIGURATION_FILE
+                    )
+
+# }}}
+
+
+# When {{{
+
+@when(parsers.parse("we import a Solution archive '{path}'"))
+def import_solution(request, host, path):
+    iso_root = request.config.getoption("--iso-root")
+    cmd = str(iso_root / "solutions.sh")
+    with host.sudo():
+        res = host.run(' '.join([
+            cmd,
+            'import --archive',
+            '{}'.format(path),
+        ]))
+        assert res.rc == 0, res.stdout
+
+
+@when(parsers.parse("I activate Solution '{name}' with version '{version}'"))
+def activate_example_solution(request, host, name, version):
+    iso_root = request.config.getoption("--iso-root")
+    cmd = str(iso_root / "solutions.sh")
+    with host.sudo():
+        res = host.run(' '.join([
+            cmd,
+            'activate --name {}'.format(name),
+            '--version {}'.format(version),
+        ]))
+        assert res.rc == 0, res.stdout
+
+
+@when(parsers.parse(
+    "I create a solution environment '{environment}' for Solution '{name}'"))
+def create_solution_environment(request, host, environment, name):
+    iso_root = request.config.getoption("--iso-root")
+    cmd = str(iso_root / "solutions.sh")
+    with host.sudo():
+        res = host.run(' '.join([
+            cmd,
+            'create-env --name {}'.format(environment),
+        ]))
+        assert res.rc == 0, res.stdout
+
+
+@when("we remove Taints on Bootstrap node before deployment")
+def remove_bootstrapNode_taints(host):
+    # Remove Taints on the Bootstrap node before deploying
+    # example-solution-operator else, the operator pods will not startup
+
+    with host.sudo():
+        res = host.run(' '.join([
+            'kubectl --kubeconfig=/etc/kubernetes/admin.conf ',
+            'taint nodes --all {}'.format(BOOTSTRAP_TAINT),
+        ]))
+        assert res.rc == 0, res.stdout
+
+    with host.sudo():
+        res = host.run(' '.join([
+            'kubectl --kubeconfig=/etc/kubernetes/admin.conf ',
+            'taint nodes --all {}'.format(INFRA_TAINT),
+        ]))
+        assert res.rc == 0, res.stdout
```

Comment:
Why using 2 times `host.sudo()` and why are you using list if you're not really splitting arguments ?

Reply:
Lets shoot for the split argument version